runelite-client/src/main/java/net/runelite/client/plugins/randomeventnotifier/RandomEventNotifier.java

```diff
@@ -0,0 +1,144 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *	list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *	this list of conditions and the following disclaimer in the documentation
+ *	and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.randomeventnotifier;
+
+import com.google.inject.Provides;
+import java.util.Arrays;
+import java.util.List;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.*;
+import net.runelite.api.events.GameTick;
+import net.runelite.client.Notifier;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.api.events.NpcSpawned;
+
+
+@PluginDescriptor(
+		name = "Random Event Notifier",
+		description = "Send a notification when a random event has spawned",
+		tags = {"random", "notifications", "event"}
+)
+@Slf4j
+public class RandomEventNotifier extends Plugin
+{
+
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private RandomEventNotifierConfig config;
+
+	private NPC lastRandomEventNpc;
+
+	@Provides
+	RandomEventNotifierConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(RandomEventNotifierConfig.class);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (randomEventSpawned())
+		{
+			notifier.notify("[" + lastRandomEventNpc.getName() + "] has spawned!");
+			lastRandomEventNpc = null;
+		}
+	}
+
+	public boolean randomEventSpawned()
+	{
+		if (lastRandomEventNpc != null) return true;
+		return false;
+	}
+
+	@Subscribe
+	public void onNpcSpawned(NpcSpawned event)
+	{
+		final NPC npc = event.getNpc();
+		List<String> npcActions = Arrays.asList(npc.getComposition().getActions());
+		if (isRandomEventNPC(npc.getId()) && npcActions.contains("Dismiss"))
+		{
+			lastRandomEventNpc = npc;
```

Comment:
I would probably remove
`
	@Subscribe
	public void onGameTick(GameTick event)
	{
		if (randomEventSpawned())
		{
			notifier.notify("[" + lastRandomEventNpc.getName() + "] has spawned!");
			lastRandomEventNpc = null;
		}
	}

	public boolean randomEventSpawned()
	{
		if (lastRandomEventNpc != null) return true;
		return false;
	}
`

And just handle everything in the OnNpcSpawned event.

Reply:
I also think this plugin would not meet quality standards if it sends a notification even when a randomevent spawns for another player then the one using the plugin.

Reply:
@robinwithes With the new overhead text implementation, I think keeping the on game tick check is necessary as the overhead text might not happen on the same game tick as the spawn event.